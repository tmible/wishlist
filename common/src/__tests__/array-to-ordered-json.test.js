import { describe, expect, it } from 'vitest';
import arrayToOrderedJSON from '../array-to-ordered-json.js';

describe('arrayToOrderedJSON', () => {
  it('should convert array to ordered JSON', () => {
    expect(
      arrayToOrderedJSON([{
        property1: 0,
        property2: 0,
        extraProperty1: 'extra',
      }, {
        property1: 0,
        property2: 1,
        extraProperty2: 'extra',
      }, {
        property1: 1,
        property2: 0,
      }, {
        property1: 1,
        property2: 1,
      }, {
        property1: 1,
        property2: 1,
      }].sort(() => Math.random() - 0.5)),
    ).toMatchSnapshot();
  });
});
