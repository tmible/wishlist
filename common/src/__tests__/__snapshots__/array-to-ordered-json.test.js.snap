// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`arrayToOrderedJSON > should convert array to ordered JSON 1`] = `"[{"extraProperty1":"extra","property1":0,"property2":0},{"extraProperty2":"extra","property1":0,"property2":1},{"property1":1,"property2":0},{"property1":1,"property2":1},{"property1":1,"property2":1}]"`;
